src/Datadog.Trace/Agent/Transports/HttpStreamRequest.cs

```diff
@@ -43,17 +43,19 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> traces)
                 // send request, get response
                 var response = await _client.SendAsync(request, bidirectionalStream, bidirectionalStream).ConfigureAwait(false);
 
-                // buffer the entire contents for now
-                var responseContentStream = new MemoryStream();
-                await response.Content.CopyToAsync(responseContentStream, ResponseReadBufferSize).ConfigureAwait(false);
-                responseContentStream.Position = 0;
-
-                var contentLength = response.ContentLength;
-                if (contentLength != null && contentLength != responseContentStream.Length)
+                // Content-Length is required as we don't support chunked transfer
+                var contentLength = response.Content.Length;
+                if (!contentLength.HasValue)
                 {
-                    throw new Exception("Content length from http headers does not match content's actual length.");
+                    throw new Exception("Content-Length is required but was not provided");
                 }
 
+                // buffer the entire contents for now
+                var buffer = new byte[contentLength.Value];
+                var responseContentStream = new MemoryStream(buffer);
+                await response.Content.CopyToAsync(buffer).ConfigureAwait(false);
+                responseContentStream.Position = 0;
```

Comment:
It seems like we could have handled both cases (copying from request content and from response content) with the `IHttpContent.CopyToAsync(Stream, ...)` overload. Here we could have passed the `MemoryStream` instead of a byte array as the destination. What do you think?

Reply:
Something like this:

```csharp
                var responseContentStream = new MemoryStream(contentLength.Value);
                await response.Content.CopyToAsync(contentLength.Value, null).ConfigureAwait(false);
                responseContentStream.Position = 0;
```

... and then in `StreamContent`, move the logic to copy a specific number of bytes from `CopyToAsync(byte[])` to `CopyToAsync(Stream, ...)` (as it used to be originally).

Reply:
>and then in StreamContent, we could move the logic that copies a specific number of bytes from `CopyToAsync(byte[])` to `CopyToAsync(Stream, ...)`.

Unfortunately, there's no API for copying a fixed number of bytes from one stream to another. We _could_ create another buffer in `CopyToAsync()` , so we copy bytes from response stream -> buffer -> parameter stream, but that seems like an extra copy for no real good reason?

After the changes in https://github.com/DataDog/dd-trace-dotnet/pull/1218/commits/eb26be51549c55856ae87a3d420116f1dfe696bc, we no longer ever use the `bufferSize` parameter so I've removed that in https://github.com/DataDog/dd-trace-dotnet/pull/1218/commits/8d8d1429ad5cc242f1b19352513ffb45da091876, but I think the `byte[]` approach is still the simplest here @lucaspimentel?

Reply:
>but that seems like an extra copy for no real good reason

Yeah, good point. Thanks for looking into it, though.